test/hwdb-test.sh

```diff
@@ -32,11 +32,9 @@ D=$(mktemp --directory)
 trap "rm -rf '$D'" EXIT INT QUIT PIPE
 mkdir -p "$D/etc/udev"
 ln -s "$ROOTDIR/hwdb" "$D/etc/udev/hwdb.d"
-err=$("$SYSTEMD_HWDB" update --root "$D" 2>&1 >/dev/null)
-if [ -n "$err" ]; then
-    echo "$err"
-    exit 1
-fi
+
+"$SYSTEMD_HWDB" update --root "$D" 2>&1 >/dev/null
+
```

Comment:
Well, maybe exiting with zero is intended to not completely break package updates or something such. So I think test/hwdb-test.sh should fail either on non-zero exit code or on non-empty stderr.